ansible/roles/k8s_haproxy/tasks/main.yml

```diff
@@ -20,3 +20,75 @@
 - name: Restart HAProxy
   become: yes
   service: name=haproxy state=restarted
+
+- name: Allow necessary ports for ufw
+  become: yes
+  ufw:
+    rule: allow
+    port: '{{ item }}'
+  with_items: 
+    - '{{ http_port }}'
+    - '{{ ssl_port }}'
+    - '{{ ssh_port }}'
+    - '{{ k8s_port }}'
+
+- name: Enable ufw
+  become: yes
+  ufw:
+    state: enabled
+    policy: deny
+
+- name: Install Apache
+  apt: name=apache2 update_cache=yes state=latest
+  environment: "{{ proxy_env | default({}) }}" 
+
+- name: Create document root
+  file:
+    path: "/var/www/html"
+    state: directory
+    mode: '777'
```

Comment:
>777 [](start = 11, length = 3)

Is it too broad permission?

Reply:
I changed to 755 permissions. This was the most restrictive I could make it while ensuring that the SONiC DUT could still download the file properly. This is also in-line with the response here https://askubuntu.com/questions/451922/apache-access-denied-because-search-permissions-are-missing